src/Microsoft.PowerShell.ConsoleGuiTools/ConsoleGui.cs

```diff
@@ -118,14 +124,37 @@ public void Start(ApplicationData applicationData)
             var list = new ListView(items)
             {
                 X = 3,
-                Y = 3,
+                Y = 1,
                 Width = Dim.Fill(2),
-                Height = Dim.Fill(2),
+                Height = Dim.Fill(0),
                 AllowsMarking = applicationData.PassThru
             };
             
             win.Add(list);
 
+            win.Enter_Pressed += () =>
+            {
+                // Regardless of whether the item was already marked, if the user presses
+                // ENTER, mark that item. This feels the most intuitive. The alternatives are:
+                // (a) Assume the user means to choose the item that's selected and mark it.
+                // (b) Act like ESC except Pass Through an marked items.
+                // (c) Toggle the selected item.
+                // Option (c) violates the principle of least astonishment: If the user has an item marked, and selected
+                // and hits ENTER s/he would be surprised to see it not returned. Esp in the single choice (most common) case.
+                // Option (b) means in the most common case (a single item is wanted) ENTER does nothing. Again surprising.
+                // It seems (a) is the most intuitive so that's what's implemented.
+                list.Source.SetMark(list.SelectedItem, true);
```

Comment:
I think it is not right behavior. If I go through an item list and select some items by mouse, cursor is on not selected item and I press Enter. In the case I unexpectedly get the last item in result list. 
I believe Enter action should be exclusively `RequestStop()`.
And I'd expect that we explicitly select items by SPACEBAR (or mouse click).

Reply:
I'd agree with you if the mouse was supported. But you can't select items using `gui.c`. 

I think this is the most natural behavior for a keyboard-only UI.

Reply:
Oops. I just realized the mouse DOES work with `gui.cs`. It just doesn't work in Windows Terminal! (It works in `cmd.exe`). 

I now think you're right.